src/Sylius/Bundle/OrderBundle/Controller/OrderItemController.php

```diff
@@ -241,4 +252,46 @@ protected function getCartManager()
     {
         return $this->get('sylius.manager.order');
     }
+
+    /**
+     * @param OrderItemInterface $orderItem
+     *
+     * @return ConstraintViolationListInterface
+     */
+    private function getCartItemErrors(OrderItemInterface $orderItem)
+    {
+        return $this
+            ->get('validator')
+            ->validate($orderItem, null, $this->getParameter('sylius.form.type.order_item.validation_groups'))
+        ;
+    }
+
+    /**
+     * @param ConstraintViolationListInterface $errors
+     * @param FormInterface $form
+     *
+     * @return FormInterface
+     */
+    private function getAddToCartFormWithErrors(ConstraintViolationListInterface $errors, FormInterface $form)
+    {
+        foreach ($errors as $error) {
+            $form->get('cartItem')->get($error->getPropertyPath())->addError(new FormError($error->getMessage()));
+        }
+
+        return $form;
+    }
+
+    /**
+     * @param RequestConfiguration $configuration
+     * @param FormInterface $form
+     *
+     * @return mixed
```

Comment:
Either this needs to be changed to `Response` or [this](https://github.com/Sylius/Sylius/pull/7548/files#diff-381ec1aa6cbf13ba994bb6bcbb539e2cL39), to mixed.